tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,220 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR Condition OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	. "github.com/istio-ecosystem/sail-operator/pkg/util/tests/ginkgo"
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/helm"
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/kubectl"
+	resourcecondition "github.com/istio-ecosystem/sail-operator/pkg/util/tests/types"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var (
+		resourceAvailable = resourcecondition.Condition{
+			Type:   "Available",
+			Status: "True",
+		}
+		resourceReconcilied = resourcecondition.Condition{
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = resourcecondition.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+
+		crds = []string{
+			// TODO: Find an alternative to this list
+			"authorizationpolicies.security.istio.io",
+			"destinationrules.networking.istio.io",
+			"envoyfilters.networking.istio.io",
+			"gateways.networking.istio.io",
+			"istiorevisions.operator.istio.io",
+			"istios.operator.istio.io",
+			"peerauthentications.security.istio.io",
+			"proxyconfigs.networking.istio.io",
+			"requestauthentications.security.istio.io",
+			"serviceentries.networking.istio.io",
+			"sidecars.networking.istio.io",
+			"telemetries.telemetry.istio.io",
+			"virtualservices.networking.istio.io",
+			"wasmplugins.extensions.istio.io",
+			"workloadentries.networking.istio.io",
+			"workloadgroups.networking.istio.io",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("installed via helm install", func() {
+			BeforeAll(func() {
+				if skipDeploy {
+					Skip("Skipping the Test as SKIP_DEPLOY is set to true")
```

Comment:
Now I'm really confused. Does this skip the entire test or just the deployment of the operator? If it's the entire test, then why is the variable called `skipDeploy`?

Reply:
The skip only does the skip of the node where it's, so only will skip the entire When node, I can change the name of the var to something more specific like skipInstallation or something similar more clear. wdyt?

Reply:
If the point is to skip the deployment of the operator, then this should print "skipping the deployment of the operator", not "skipping the test".

Reply:
Also, this should be one level higher then.. directly under `Describe("installation"...`.